File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool match(char c1, char c2){
        if(c1 == '(' && c2 == ')') return true;
        if(c1 == '{' && c2 == '}') return true;
        if(c1 == '[' && c2 == ']') return true;
        else return false;
    }
    bool isValid(string s) {
        int n = s.length();
        stack<char> st;
        for(int i = 0;i<n;i++){
            if(s[i] == '(' || s[i] == '{' || s[i] == '[') st.push(s[i]);
            else{
                if(st.empty() || !match(st.top(), s[i])) return false;
                else st.pop();
            }
        }
        if(st.empty()) return true;
        return false;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> st = new Stack<char>();
        
        for(int i = 0; i < s.Length; i++){
            if(s[i] == '(' || s[i] == '{' || s[i] == '['){
                st.Push(s[i]);
            }
            else{
                char c = s[i];
                switch(c){
                    case ')':
                        if(st.Count() > 0 && st.Peek() == '(') st.Pop();
                        else return false;
                        break;
                    case '}':
                        if(st.Count() > 0 && st.Peek() == '{') st.Pop();
                        else return false;
                        break;
                    case ']':
                        if(st.Count() > 0 && st.Peek() == '[') st.Pop();
                        else return false;
                        break;
                    default:
                        return false;
                }
            }
        }
        if(st.Count() == 0) return true;
        return false;
    }
}